cmd/buildah/umount.go

```diff
@@ -10,44 +10,73 @@ import (
 )
 
 var (
+	umountFlags = []cli.Flag{
+		cli.BoolFlag{
+			Name:  "all, a",
+			Usage: "umount all of the currently mounted containers",
+		},
+	}
 	umountCommand = cli.Command{
 		Name:           "umount",
 		Aliases:        []string{"unmount"},
 		Usage:          "Unmounts the root file system on the specified working containers",
 		Description:    "Unmounts the root file system on the specified working containers",
 		Action:         umountCmd,
 		ArgsUsage:      "CONTAINER-NAME-OR-ID [...]",
+		Flags:          umountFlags,
 		SkipArgReorder: true,
 	}
 )
 
 func umountCmd(c *cli.Context) error {
+	umountAll := c.Bool("all")
 	umountContainerErrStr := "error unmounting container"
 	args := c.Args()
-	if len(args) == 0 {
+	if len(args) == 0 && !umountAll {
 		return errors.Errorf("at least one container ID must be specified")
 	}
+	if len(args) > 0 && umountAll {
+		return errors.Errorf("when using the --all switch, you may not pass anycontainers IDs")
+	}
 
 	store, err := getStore(c)
 	if err != nil {
 		return err
 	}
 
 	var lastError error
-	for _, name := range args {
+	if len(args) > 0 {
+		for _, name := range args {
+			builder, err := openBuilder(getContext(), store, name)
+			if err != nil {
+				lastError = util.WriteError(os.Stderr, errors.Wrapf(err, "%s %s", umountContainerErrStr, name), lastError)
```

Comment:
You need to check if lastError is already set and output it before replacing the lastError with the new one.
```
if lastError != nil {
           logrus.Error(lastError)
}
```